### PR TITLE
Add summit data endpoint

### PR DESCRIPTION
**Pull request by jmjcobs**
- add koa-route middleware from [https://github.com/koajs/route](url)
- setup endpoint in server for summit data api
- Resolve Travis error unexpected require() global-require.  ES lint throws error when require () is in an arrow function.  Removed require() from arrow function and placed in top-level scope.  https://eslint.org/docs/rules/global-require

### CODE REVIEW

File: server/index.js
Comment (dangodev) [inline server/index.js]: This is actually a good example of fighting with ESLint, and whether it’s helping or hurting. In general, it’s better to do as it says, but this case is different.

Here, we’re loading a file in-memory, before the endpoint. What if a user doesn’t visit the API endpoint? That means the server loaded something in memory it didn’t need.

There are actually 3 ways to modify ESLint behavior:
1. Modify `.eslintrc` (this becomes a global rule though)
2. Use `/* eslint-disable RULE */` (here, it’d be `/* eslint-disable global-require */`). This disables the rule in the current file until a `/* eslint-enable RULE */` is reached.
3. Use `// eslint-disable-next-line global-require`. This applies to the next line only.

In general I prefer #3 when I like the rule, but I’m making an exception because I’m smarter.

So, yeah, you only need this 1% of the time, but you get the best of both worlds: you get great linting but can override it when you need to!

File: server/index.js
Comment (dangodev) [inline server/index.js]: Another note on arrow functions: These two are the same:

```js
app.use(get( … )) // implicit return: get() passes its value to app.use()
```

```js
app.use(() => {
  return get( … ) // explicit return through a function: get() passes its value to our anonymous function, which then passes its value to app.use()
}
```

In #2, we’re inserting an extra function, but we’re returning the value, so it’s the same. #2 would be pretty dumb—it doesn’t do anything—but it may be helpful to think you always have this option.

`get( … )` is built to return a value, which needs to get returned somewhere that uses it.

---

It may be helpful to ask yourself for every function: _what is this returning_?

Take the internal function:

```js
  get('/api/v1/peaks', ctx => {
    ctx.body = data;
  })
```

We’re actually calling a function for the 2nd parameter. That’s the same as this:

```js
  get('/api/v1/peaks', function (ctx) {
    ctx.body = data;
  })
```

The only difference is that the arrow function just saves us space (there are other minor differences, but don’t worry about that now). Notice that **we’re not returning anything.** Now, to me, this is weird. But I’m just following the examples so I assume everything works all right, and this PR works. So we can assume that that `ctx` value—even though we’re not returning it—is tracked and taken care of within our `koa-route` package perfectly. So we can just not think about it!

But anyways, the point is to just focus on `return` statements—explicit or implicit. Not sure what’s returning? `console.log` it! If it’s returning nothing, it’ll show `undefined`.

**Review by drwpow — APPROVED** [review]
looks good to me 👌

File: package.json
Comment (dangodev) [inline package.json]: It looks like we don’t need this. It’s not hurting anything, but probably best to remove it at some point: `npm uninstall --save koa-parse`

File: server/index.js
Comment (jmjcobs) [reply server/index.js]: Lot's of good tips in here.  Thanks! 

What's the best practice from a version control standpoint on these recommendations?  Is it best to delete the commit before merging, modify the files and then add a new cleaned up commit?

Comment (dangodev) [reply server/index.js]: So for this, I ✅’d it so you can merge if you want to. From here, it’s your call. If I wasn’t happy with it, I would have requested changes.

But in terms of Git, you can make an additional commit to this branch specifying what you did. Like, `Remove koa-parse`, etc. And just push onto this PR.

When you’re happy with it, merge it!

**Pull request merged**
